src/lib/models/item.ts

```diff
@@ -0,0 +1,52 @@
+import { z } from 'zod'
+import { fromError } from 'zod-validation-error'
+import prisma from '../prisma'
+import { AuthUser } from './user'
+
+type ListItem = {
+  id: string
+  name: string
+}
+
+type ListItemDetails = {
+  id: string
+  name: string
+  description: string | null
+}
+
+export const listMyItems = async (authUser: AuthUser): Promise<ListItem[]> => {
```

Comment:
This is the "corest" file of the entire application. It is domain logic that is agnostic to react, nextjs, auth libraries, etc. It is just business logic, validation, and database interactions. These model functions can be called independent of many third party tool decisions. 

I think it is important to keep these types of thing isolated from the rest of the app because it is clear and easy to understand where to put stuff, and also makes it much easier to make big changes in the future.